L1-CPP5term-Calculator/Calculator.cpp

```diff
@@ -0,0 +1,112 @@
+#include "Calculator.h"
+
+template <typename T1, typename T2>
+bool convertToT(T1& t1, T2& t2){
+    std::stringstream s;
+    s<<t1;
+    return static_cast<bool>(s >> t2);
+}
+
+Calculator::Calculator() {
+    precedence["+"] = 1;
+    precedence["-"] = 1;
+    precedence["*"] = 2;
+    precedence["/"] = 2;
+    precedence["^"] = 3;
+
+    functions["+"] = [](double a, double b)->double {return b + a; };
+    functions["-"] = [](double a, double b)->double {return b - a; };
+    functions["*"] = [](double a, double b)->double {return b * a; };
+    functions["/"] = [](double a, double b)->double {return b / a; };
+    functions["^"] = [](double a, double b) {return pow(b, a); };
+}
+
+std::string Calculator::to_default_view(std::string str) {
+    std::string def;
+    std::string t;
+    double d;
+    char previous = '(';
+    for (auto u : str){
+        if (isspace(u)) continue;
+        convertToT(u, t);
+
+        if (u == '-' && previous == '('){
+            def += " 0 " + t;
+            previous = u;
+            continue;
+        }
+
+        if (functions.contains(t) || t == "(" || t == ")"){
+            def += " " + t + " ";
+            previous = u;
+            continue;
+        }
+
+        if (convertToT(u, d) != convertToT(previous, d) && u != '.' && previous != '.') {

```

Comment:
Here type conversion is confused with value conversion (naming could fix that) Don't you try to extract float from stream in a char-by-char fashion?